judo-requirement-report-test/src/test/java/hu/blackbelt/judo/requirement/test/TestRequirementProcessor.java

```diff
@@ -0,0 +1,238 @@
+package hu.blackbelt.judo.requirement.test;
+
+/*-
+ * #%L
+ * JUDO Requirement :: Report :: Test
+ * %%
+ * Copyright (C) 2018 - 2023 BlackBelt Technology
+ * %%
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License, v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is
+ * available at https://www.gnu.org/software/classpath/license.html.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
+import com.opencsv.CSVParser;
+import com.opencsv.CSVParserBuilder;
+import com.opencsv.CSVReaderBuilder;
+import hu.blackbelt.judo.requirement.report.annotation.Requirement;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.opencsv.CSVReader;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Slf4j
+public class TestRequirementProcessor {
+
+    @Requirement(reqs = {
+
+    })
+    @Test
+    public void test00(){
+        assertTrue(true);
+    }
+
+    @Requirement(reqs = {
+            "R01"
+    })
+    @Test
+    public void test01(){
+        assertTrue(true);
+    }
+
+    @Requirement(reqs = {
+            "R02"
+    })
+    @Test
+    public void test02(){
+        assertTrue(true);
+    }
+
+    @Requirement(reqs = {
+            "R01",
+            "R03"
+    })
+    @Test
+    public void test03(){
+        assertTrue(true);
+    }
+
+    @Requirement(reqs = {
+            "R05",
+            "R04",
+            "R01"
+    })
+    @Test
+    public void test04(){
+        assertTrue(true);
+    }
+
+    @Requirement(reqs = {
+            "R01",
+            "R03"
+    })
+    /**
+     * This is not a real test. Therefore, it isn't used the @Test annotation.
+     */
+    public void test05(){
+        assertTrue(false);
+    }
+
+    @Test
+    /**
+     * This is the real test case.
+     */
+    public void testReal(){
+        // Create Reference Array
+        List<String[]> refTable = createRefTable();
+
+        // Read generated csv
+        List<String[]> table = readCsv(System.getProperty("reportPath"));
+
+        // Check the lists
+        assertArrayEquals(refTable.toArray(), table.toArray());
```

Comment:
Optional recommendation

When you are using hamcrest matcher for collection, the createRefTable and comparator can be avoided.

```
assertThat(myList , contains(allOf(hasProperty("id", is(7L)), 
                                   hasProperty("name", is("testName1")),
                                   hasProperty("description", is("testDesc1"))),
                             allOf(hasProperty("id", is(11L)), 
                                   hasProperty("name", is("testName2")),
                                   hasProperty("description", is("testDesc2")))));
```

Reply:
I have rewritten. :-)